code/Control/UserDefinedFormController.php

```diff
@@ -268,6 +274,12 @@ public function process($data, $form)
                             return;
                         }
 
+                        // generate image thumbnail to show in asset-admin
+                        // you can run userforms without asset-admin, so need to ensure asset-admin is installed
+                        if (class_exists(AssetAdmin::class) && $file->getIsImage()) {
+                            Injector::inst()->get(AssetAdmin::class)->generateThumbnails($file);
```

Comment:
I realise people may not be a fan of `class_exists(AssetAdmin::class)`

Other options considered:

a) extension hook `$this->extend('updateFileUpload', $file)`
- Don't think we'd want to create a `UserDefinedFormControllerExtension.php` in asset-admin

b) make asset-admin a composer requirement of userforms
- would mean asset-admin is installed on lightweight sites that may not want it

c) Generate the thumbnails in FormField
- would generate asset-admin thumbnails for legacy forms
- however framework does not include asset-admin either
- may create regressions elsewhere, or undesired thumbnail creation

Reply:
Posted another suggestion on https://github.com/silverstripe/silverstripe-assets/issues/399#issuecomment-631997585

d) Add an extension hook on framework/../FormField and extension method in asset-admin/../FormFieldExtension
- having a central place to generate thumbnails means we'll also need to remove any other thumbnail generation done elsewhere in the site, notably on upload within asset-admin https://github.com/silverstripe/silverstripe-asset-admin/blob/1/code/Controller/AssetAdmin.php#L1172